### PR TITLE
Add and fix missing `Triple` tests

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Vendored Swift Driver triple was not checked against a few tests that weren't brought over from TSC after `TSC.Triple` type was deprecated. We should fix those tests, especially as they verified that per-component equality for triples worked instead of the current string-based equality check.

Also fixed some of the `Utilities/soundness.sh` script failures.

Related to rdar://113967401

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by neonichu** [comment]
Thanks! 

The changes to vendored files are only `swiftformat` related, right?

**Comment by MaxDesiatov** [comment]
> The changes to vendored files are only `swiftformat` related, right?

Not exclusively, let me revert the formatting changes to make it easier to review those.

**Comment by MaxDesiatov** [comment]
@neonichu ready for review

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test linux

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
I prefer to upstream these changes, will create corresponding PRs on swift-driver after this one is merged.

**Comment by neonichu** [comment]
> I prefer to upstream these changes, will create corresponding PRs on swift-driver after this one is merged.

Could we do that in reverse order for the `main` PR? I think only the 5.9 PR is urgent, so we can just wait on feedback from the driver team for the `main` one. Just trying to avoid having to change this again if there's feedback from them that materially changes what is happening in this PR.

**Comment by MaxDesiatov** [comment]
Makes sense, I need a review on the 5.9 PR then so that it unblocks versioned triples checks in #6819. Thanks!

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by neonichu — APPROVED** [review]

**Pull request merged**
